<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainDialog</class>
 <widget class="QDialog" name="MainDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>533</width>
    <height>413</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>광물시장 트레이딩</string>
  </property>
  <widget class="QGroupBox" name="learningGroupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>161</width>
     <height>71</height>
    </rect>
   </property>
   <property name="title">
    <string>강화학습방식 설정</string>
   </property>
   <widget class="QRadioButton" name="selectPPO">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>PPO</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="selectDQN">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>30</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>DQN</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="tradingGroupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>100</y>
     <width>161</width>
     <height>301</height>
    </rect>
   </property>
   <property name="title">
    <string>사용할 베이스 트레이딩 전략설정</string>
   </property>
   <widget class="QListWidget" name="tradingList">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>141</width>
      <height>261</height>
     </rect>
    </property>
    <property name="selectionMode">
     <enum>QAbstractItemView::MultiSelection</enum>
    </property>
    <item>
     <property name="text">
      <string>MACD</string>
     </property>
    </item>
   </widget>
  </widget>
  <widget class="QGroupBox" name="subjectGroupBox">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>10</y>
     <width>161</width>
     <height>301</height>
    </rect>
   </property>
   <property name="title">
    <string>트레이딩에 사용할 종목 선택</string>
   </property>
   <widget class="QListWidget" name="subjectList">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>141</width>
      <height>261</height>
     </rect>
    </property>
    <property name="selectionMode">
     <enum>QAbstractItemView::MultiSelection</enum>
    </property>
    <item>
     <property name="text">
      <string>Gold</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Silver</string>
     </property>
    </item>
   </widget>
  </widget>
  <widget class="QPushButton" name="newModelBtn">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>10</y>
     <width>181</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>모델 생성</string>
   </property>
  </widget>
  <widget class="QLabel" name="modelInfoLabel">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>320</y>
     <width>341</width>
     <height>71</height>
    </rect>
   </property>
   <property name="text">
    <string>현재 로드된 모델 파일 : 
없음</string>
   </property>
   <property name="textFormat">
    <enum>Qt::AutoText</enum>
   </property>
  </widget>
  <widget class="QPushButton" name="loadModelBtn">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>60</y>
     <width>181</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>모델 불러오기</string>
   </property>
  </widget>
  <widget class="QPushButton" name="saveModelBtn">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>110</y>
     <width>181</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>모델 저장</string>
   </property>
  </widget>
  <widget class="QPushButton" name="trainingBtn">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>160</y>
     <width>181</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>모델 학습</string>
   </property>
  </widget>
  <widget class="QPushButton" name="testBtn">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>210</y>
     <width>181</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string>가상 트레이딩</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>selectPPO</sender>
   <signal>clicked()</signal>
   <receiver>MainDialog</receiver>
   <slot>ppoSelect()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>49</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>selectDQN</sender>
   <signal>clicked()</signal>
   <receiver>MainDialog</receiver>
   <slot>dqnSelect()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>126</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>216</x>
     <y>52</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tradingList</sender>
   <signal>itemSelectionChanged()</signal>
   <receiver>MainDialog</receiver>
   <slot>changedTrading()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>249</y>
    </hint>
    <hint type="destinationlabel">
     <x>265</x>
     <y>271</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newModelBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainDialog</receiver>
   <slot>createModel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>34</y>
    </hint>
    <hint type="destinationlabel">
     <x>276</x>
     <y>321</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>loadModelBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainDialog</receiver>
   <slot>loadModel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>77</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>289</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveModelBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainDialog</receiver>
   <slot>saveModel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>130</y>
    </hint>
    <hint type="destinationlabel">
     <x>193</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>trainingBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainDialog</receiver>
   <slot>modelTraining()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>211</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>187</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>testBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainDialog</receiver>
   <slot>modelTest()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>234</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>344</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>dqnSelect()</slot>
  <slot>ppoSelect()</slot>
  <slot>changedTrading()</slot>
  <slot>changedSubject()</slot>
  <slot>createModel()</slot>
  <slot>loadModel()</slot>
  <slot>saveModel()</slot>
  <slot>modelTraining()</slot>
  <slot>modelTest()</slot>
 </slots>
</ui>
